src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/EditAndContinue/EditAndContinueProvider.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.VS;
+using EncInterop = Microsoft.VisualStudio.LanguageServices.Implementation.EditAndContinue.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Managed.VS.EditAndContinue
+{
+
+    [Export(ExportContractNames.VsTypes.ProjectNodeComExtension)]
+    [AppliesTo(ProjectCapability.ManagedEditAndContinue)]
+    [ComServiceIid(typeof(IVsENCRebuildableProjectCfg))]
+    [ComServiceIid(typeof(EncInterop.IVsENCRebuildableProjectCfg2))]
+    [ComServiceIid(typeof(EncInterop.IVsENCRebuildableProjectCfg4))]
+    class EditAndContinueProvider : IVsENCRebuildableProjectCfg, EncInterop.IVsENCRebuildableProjectCfg2, EncInterop.IVsENCRebuildableProjectCfg4
+    {
+        private const int E_NOTIMPL = unchecked((int)0x80004001);
+        private readonly ILanguageServiceHost _host;
+
+        [ImportingConstructor]
+        public EditAndContinueProvider(
+            ILanguageServiceHost host)
+        {
+            _host = host;
+        }
+
+        #region IVsENCRebuildableProjectCfg Implementation
```

Comment:
I actually like the regions because I don't have to look at some big chunk of code instead the implementation is separate and makes it look clean